metricbeat/beater/reload.go

```diff
@@ -0,0 +1,252 @@
+package beater
+
+import (
+	"expvar"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var (
+	debugr        = logp.MakeDebug("reloader")
```

Comment:
it is debugr on purpose, as debugf is already taken from metricbeat.go. But I wanted the debugging in a different namespace. Indirectly implies that later this should be move to its own namespace but this requires some more refactoring which should be done in a later step.